syntax = "proto3";
import "utils.proto";

message ModifyKarmaRequest {
  int32 user_id = 1;
  int32 delta = 2;
}

message ModifyKarmaResponse {
  OperationStatus status = 2;
}

message KarmaRequest {
  int32 user_id = 1;
}

message KarmaResponse {
  int32 karma = 1;
  OperationStatus status = 2;
}

message AddKarmaUserRequest {
  int32 karma = 1;
}

message AddKarmaUserResponse {
  int32 user_id = 1;
  OperationStatus status = 2;
}

message ChooseUsersRequest {
  repeated int32 forbidden_user_ids = 1;
  uint32 users_to_choose = 2;
}

message ChooseUsersResponse {
  repeated int32 user_ids = 1;
  OperationStatus status = 2;
}

service Karma {
  rpc ModifyKarma (ModifyKarmaRequest) returns (ModifyKarmaResponse);
  rpc GetKarma (KarmaRequest) returns (KarmaResponse);
  rpc AddUser (AddKarmaUserRequest) returns (AddKarmaUserResponse);
  rpc ChooseKarmaWeightedRandomUsers (ChooseUsersRequest) returns (ChooseUsersResponse);
}